### PR TITLE
[clang-offload-bundler] Remove support for unbundling partially linked objects

### PR DESCRIPTION
**Pull request by sndmitriev**
Use of partial linking for offload on Linux has recently been removed in the
driver (please see https://github.com/intel/llvm/pull/2867), so there are no
remaining uses of "oo" unbundling mode anymore. This patch cleans up support
for this mode to reduce amount of differences between intel/llvm and llvm.org.

Signed-off-by: Sergey Dmitriev <serguei.n.dmitriev@intel.com>

### CODE REVIEW

**Review by mdtoguchi — APPROVED** [review]
Nice cleanup, LGTM

**Comment by sndmitriev** [comment]
ping

File: clang/tools/clang-offload-bundler/ClangOffloadBundler.cpp [hunk 1]
```diff
@@ -448,96 +442,27 @@ class TempFileHandlerRAII {
...
 ///
 /// To unbundle, we just copy the contents of the designated section.
-///
-/// The bundler produces object file in host target native format (e.g. ELF for
-/// Linux). The sections it creates are:
-///
-/// <OFFLOAD_BUNDLER_MAGIC_STR><target triple 1>
-/// |
-/// | binary data for the <target 1>'s bundle
-/// |
```
Comment (kbobrovs) [inline clang/tools/clang-offload-bundler/ClangOffloadBundler.cpp:458]: I think this part is still valid even w/o the removed "oo" mode

File: clang/tools/clang-offload-bundler/ClangOffloadBundler.cpp [hunk 2]
```diff
@@ -448,96 +442,27 @@ class TempFileHandlerRAII {
...
-/// | binary data for the <target 1>'s bundle (from c_fat.o)
-/// <SIZE_SECTION_PREFIX><target triple 1>
-/// | size of the <target1>'s bundle (8 bytes) (from a_fat.o)
-/// | size of the <target1>'s bundle (8 bytes) (from b_fat.o)
-/// | size of the <target1>'s bundle (8 bytes) (from c_fat.o)
-/// ...
-///
-/// The alignment of all the added sections is set to one to avoid padding
-/// between concatenated parts.
-///
```
Comment (kbobrovs) [inline clang/tools/clang-offload-bundler/ClangOffloadBundler.cpp:494]: this is also still valid

File: clang/tools/clang-offload-bundler/ClangOffloadBundler.cpp (hunk 1 above)
Comment (sndmitriev) [reply clang/tools/clang-offload-bundler/ClangOffloadBundler.cpp]: Added this part back as well as the next one.

**Review by kbobrovs — APPROVED** [review]

**Review by mdtoguchi — APPROVED** [review]

**Pull request merged**